### DISCUSSION TITLE
Having issue with configuring nginx with OSM

### DISCUSSION THREAD

**Discussion by rcpandula** [openservicemesh/osm#5243] [Q&A] [answered]
Hi, 

I have been trying to setup the nginx tls version endpoint for external access to expose a public end point on my local cluster( using the docker-desktop). 

I have sample grpc service which is deployed locally and used the nginx with self signed signed certificate. I could able to access the service using the nginx https endpoint. No issue observed. 

Now, I have installed the OSM in my local cluster, with below override configuration: 
# the original file is available https://github.com/openservicemesh/osm/blob/release-v0.11/charts/osm/values.yaml
OpenServiceMesh:
  # Enable permissive traffic policy mode
  enablePermissiveTrafficPolicy: true

  # Log level for the Envoy proxy sidecar. Non developers should generally never set this value. In production environments the LogLevel should be set to `error`
  envoyLogLevel: trace

  # Enable the debug HTTP server on OSM controlle
  enableDebugServer: true

  # When enabled, fine grained control over Egress (external) traffic is enforced
  # Enable egress in the mesh
  enableEgress: true

  # -- Enable mesh-wide HTTPS ingress capability (HTTP ingress is the default)
  useHTTPSIngress: true

Then, added my namespace to the OSM, and deployed the service. Automatically it injected the envoy proxy for the sample service. 

Then i have created the ingress / ingressbackend. 

apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: demo-test
  namespace: demo-namespace
  labels:
    app: demoapp
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/backend-protocol: "GRPC"
    nginx.ingress.kubernetes.io/configuration-snippet: |
      proxy_ssl_name "demoapp.uop.cluster.local";
    nginx.ingress.kubernetes.io/proxy-ssl-secret: "osm-system/osm-nginx-client-cert"
    nginx.ingress.kubernetes.io/proxy-ssl-verify: "on"
spec:
  ingressClassName: nginx
  rules:
  - host: demoapp-test.127.0.0.1.nip.io
    http:
      paths:
      - path: /
        pathType: Prefix
        backend:
          service:
            name: demoapp
            port:
              name: grpc

---
kind: IngressBackend
apiVersion: policy.openservicemesh.io/v1alpha1
metadata:
  name: demoapp-test
  namespace: demo-namespace
spec:
  backends:
  - name: demoapp-test
    port:
      number: 8080
      protocol: https
    tls: 
      skipClientCertValidation: true
  sources:
  - kind: Service
    namespace: ingress-nginx
    name: nginx-ingress-controller
  - kind: AuthenticatedPrincipal
    name: ingress-nginx.ingress-nginx.cluster.local

But, somereasons, I am not able to access the service using the ingress end point. 

i have followed the example given @ https://release-v0-11.docs.openservicemesh.io/docs/demos/ingress_k8s_nginx/

I have been working on this issue for several days, but could not identify the rootcause... 

appreciate your help...

FYI... i have tried online for help, but not much information available. Same application is working fine without much configuration changes using istio service mesh.

**Comment by keithmattix** [comment, marked as answer]
Hi! Thanks for submitting this; an Issue is probably the best way to get some eyes on your problem as opposed to discussions. If you wouldn't mind filing this as an issue, then please resubmit and someone will take a look. Also, it would be helpful to attach any (redacted) error logs or response codes so we can better assist you

**Reply by rcpandula** [reply]
Created issue: https://github.com/openservicemesh/osm/issues/5247
